Fixing example code

ian-coccimiglio opened the pull request:
`goldenRatioInPython` needs to define `iterations`, and the returned Integers can't be cast to Long (a bug?).

ian-coccimiglio commented:
It's just a test example, but it was instructive for me to learn about Java's type system. Since `task.outputs.get("numer")` might return an `(Integer)` or a `(Long)` based on how big the numerator and denomerator are, the value needs to be appropriately unboxed and widened to a `long` before calculation. If `iterations` is greater than 47, the script runs and the ratio is printed to STDOUT. If `iterations` is smaller than 47, nothing is printed to STDOUT and the method exits silently (which is I think a bug). And, if `iterations` is greater than 91, the ratio can no longer be calculated accurately.

To handle both cases, we could use the following conversion, but it's unnecessary for a test script:

```java
private static long convert_to_long(Object value) throws IllegalArgumentException {
    if (value instanceof Integer) {
        return ((Integer) value).longValue();
    } else if (value instanceof Long) {
        return ((Long) value);
    } else {
        throw new IllegalArgumentException("Expected numeric value for 'value'");
    }
}
```

ctrueden commented:
@ian-coccimiglio Thanks for the PR! :star: 

For the long value, both boxed types `Integer` and `Long` (and `Short` and `Byte` and `Float` and `Double`) extend `Number`, so we could simplify your idea to:
```java
    if (value instanceof Number) {
        return ((Number) value).longValue();
    } else {
        throw new IllegalArgumentException("Expected numeric value for 'value'");
    }
```

Generally speaking, when you want to call a particular method (in this case `longValue()`), it can be very instructive to peek at which type in the type hierarchy defines that function, and then check for that type, since access to that type's method is all that's needed. Not so different from Python where you'd check the object for the needed attribute (function or otherwise).

ian-coccimiglio committed:
Fix the example code

The goldenRatioInPython function needs to define iterations to function,
and the returned numeric value types should be handled more defensively.

Co-authored-by: Curtis Rueden <ctrueden@wisc.edu>

ctrueden commented:
Ach, dropped the ball on finalizing this! I just rebased over the latest main branch, and adjusted the example code to be more general with the `Number` base class as mentioned above.
